### PR TITLE
Use prebuilt binaries for docker image too.

### PR DESCRIPTION
**Pull request by anuraaga**
*Issue #, if available:*

*Description of changes:*
Updates Dockerfile to copy binaries that have already been built if they exist to avoid having multiple ground-truth binaries.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by willarmiros — APPROVED** [review]

File: Tool/src/build-in-docker.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+TARGETPLATFORM=${TARGETPLATFORM:-linux/amd64}
+BUILD_FOLDER=${TARGETPLATFORM/\//-}
+
+if [[ -d "/workspace/build/xray-${BUILD_FOLDER}" ]]; then
+  echo "Copying prebuilt binary"
+  cp /workspace/build/xray-${BUILD_FOLDER}/xray /workspace
```
Comment (bhautikpip) [inline Tool/src/build-in-docker.sh:8]: Looks like Dockerfile is picking binaries from `/workspace/xray` dir then shouldn't this be `cp /workspace/build/xray-${BUILD_FOLDER}/xray /workspace/xray` ? or am I missing something here?

File: Tool/src/build-in-docker.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+else
+  echo "Building from source"
+  CGO_ENABLED=0 GOOS=linux GOARCH=$(echo $TARGETPLATFORM | cut -d'/' -f2) go build -a -ldflags '-extldflags "-static"' -o /workspace/xray /workspace/cmd/tracing
```
Comment (bhautikpip) [inline Tool/src/build-in-docker.sh:11]: Will this command also build binaries for linux/arm64 platform? Not super familiar with shell. I guess I only saw `linux/amd64`  value?

File: .github/workflows/continuous-build.yml [hunk 3]
```diff
@@ -43,23 +43,14 @@ jobs:
...
-      - name: Checkout Repository
-        uses: actions/checkout@v2
       - name: Login to DockerHub
-        if: ${{ github.event_name == 'push' }}
+        if: ${{ runner.os == 'Linux' && github.event_name == 'push' }}
         uses: docker/login-action@v1
         with:
           username: ${{ secrets.DOCKER_USER }}
           password: ${{ secrets.DOCKER_PASS }}
-      - name: Set up QEMU
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply .github/workflows/continuous-build.yml:60]: Nah - in this build we don't need qemu anymore we use the cross compiled binaries from go build.

File: Tool/src/build-in-docker.sh (hunk 1 above)
Comment (anuraaga) [reply Tool/src/build-in-docker.sh]: It is copying the file `xray` to the directory `/workspace`. I've added a trailing slash to make it more explicit

File: Tool/src/build-in-docker.sh (hunk 2 above)
Comment (anuraaga) [reply Tool/src/build-in-docker.sh:11]: Nah - this is mostly for local development or some other random need for a user to build a docker image from source when they don't have golang installed on their computer. So just `linux/amd64` is enough for it, and our full-blown build uses the prebuilt binaries that go crosscompiles in the Github action.

**Review by bhautikpip — APPROVED** [review]
Thanks for answering! Looks good :)

**Pull request merged**